06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Inverted Index class
+ */
+class InvertedIndex {
+    /**
+     * class constructor
+     */
+  constructor() {          
+    this.indices = {};
+  }
+        /**
+         * normalizedText takes a string with numbers,
+         * spaces and symbols and returns a normalized string
+         * @param{String} text - The name of the string to be normalized
+         * @return{String} normalized string
+         */
+  normalizedText(text) {
+    return text.toLowerCase().match(/[A-Za-z]+/g, (matched) => {
+      return matched.sort();
+    });
+  }
+        /**
+         * @param{String} words - String to be filtered
+         * @return{Array} token - Array without duplicate words
+         */
+  uniqueWords(words) {
+    const tokens = this.normalizedText(words);
+    return tokens.filter((item, index) =>
+                tokens.indexOf(item) === index);
+  }
+        /**
+         * createIndex takes a document and builds an idex out of it
+         * @param{String} fileName - The name of the file to be indexed
+         * @param{Array} fileContent - The content of the file to be indexed
+         * @return{Array} indices - Maps words to their location
+         */
+  createIndex(fileName, fileContent) {
+    const indexedFile = {};
+    const fileLength = fileContent.length;
+    const wordsToIndex = [];
+    if (fileLength === 0) {
```

Comment:
Other than checking for `fileContent.length` is there any other validation do on the content?

For example, to ensure it contains documents with key value pairs of `title` an `text`.